### ISSUE TITLE
Bad Request causes infinite loop

### ISSUE THREAD

**Issue by ADumaine** [mappum/bitcoin-protocol#85]
If a node is set up behind a proxy (or perhaps some other configuration) and does not return the expected response (a well formed p2p message), the decoder goes into an infinite loop.

Line 97 of types.js assumes there will be a 0 in the buffer array. 
`   for (stop = 0; bvalue[stop] !== 0; ++stop);`

A 400 response will not have a 0 in the array.

This could be easily fixed with (error checking may also need to be addressed) 
`    for (stop = 0; bvalue[stop] !== 0 && stop < 12; ++stop); `

A short term workaround is to not pipe the decoder through the socket.

```
//socket.pipe(decoder);
socket.on('data', (message) => {
        if (message.length > 132) {
          socket.end();
          console.log('Service does not appear to be a node');
          return;
        } 
        decoder.write(message);
 });
```
It would also be very helpful to document the available methods.  I had to dig through the code for the write method.

**Comment by mappum** [comment]
Good catch, I suspect there are other potential DoS vectors like this.

This project probably needs some fuzz testing or some other way of detecting bugs like this.

**Comment by ADumaine** [comment]
Yes.  It took awhile to track this down. Communicating with one particular node was crashing our servers.

**Comment by ADumaine** [comment]
Tested new version in dev and can now catch the error.  Thanks for the quick response!
